File: backend/app/schemas/query.py
"""Pydantic schemas for click-to-inspect layout query endpoint."""
from pydantic import BaseModel


class InspectElement(BaseModel):
    """A single layout instance returned by a spatial query."""

    name: str
    master: str | None
    nets: list[str]


class InspectResponse(BaseModel):
    """Response schema for GET /api/v1/query/{run_id}."""

    elements: list[InspectElement]
    run_id: str
    x_um: float
    y_um: float

File: frontend/src/api/query.ts
import { apiClient } from "./client";

export interface InspectElement {
  name: string;
  master: string | null;
  nets: string[];
}

export interface InspectResponse {
  elements: InspectElement[];
  run_id: string;
  x_um: number;
  y_um: number;
}

export async function clickToInspect(
  runId: string,
  xUm: number,
  yUm: number,
  toleranceUm: number = 1.0,
): Promise<InspectResponse> {
  const resp = await apiClient.get(`/query/${runId}`, {
    params: { x_um: xUm, y_um: yUm, tolerance_um: toleranceUm },
  });
  return resp.data;
}
